spynnaker/pyNN/models/neuron/synapse_io/synapse_io_row_based.py

```diff
@@ -61,17 +61,25 @@ def get_max_row_info(
         min_delay_for_delay_extension = (
                 max_delay_supported + numpy.finfo(numpy.double).tiny)
 
+        # get delays to pass into connector
+        delays = None
+        if hasattr(synapse_info.synapse_dynamics, 'delay'):
+            delays = synapse_info.synapse_dynamics.delay
+        else:
+            # delays come directly from the connector?
+            delays = synapse_info.connector.get_delay()
+
```

Comment:
I think this could just now use synapse_info.delay?  I don't see that the dynamics needs to be involved...  remember these are the actual delay values, not theoretical maximums or anything else that the synapse dynamics would provide.

Reply:
Looking at this a bit more, there's some strange naming going on here.  Basically in this function, synapse_dynamics_stdp is the "synapse_type" that was passed in by the user in a Projection.  I guess it's called that here (it's called "synapse_type" in the sPyNNaker8 Projection class) to avoid confusion with the synapse_type in the neuron model... anyway, this isn't really using the dynamics, it's essentially copying in what the user has defined in their Projection.

Reply:
I should clarify a bit further: here, it's called "synapse_dynamics", but at the top end where this is passed in (in pynn_projection_common) it's called "synapse_dynamics_stdp".  We should possibly come up with a whole different name instead I guess...